runtime/storage/storage-provider-base.js

```diff
@@ -62,9 +62,12 @@ export class StorageProviderBase {
     for (let [callback] of listenerMap.entries()) {
       callbacks.push(callback);
     }
+    // Yield so that event firing is not re-entrant with mutation.
+    await 0;
     for (let callback of callbacks) {
       callback(details);
     }
+    // FIXME
```

Comment:
Elaborate, please? Someone else may need to FIXTHAT and some context would be helpful :)

Reply:
Fixed it.